functions/src/has-alread-voted.ts

```diff
@@ -0,0 +1,35 @@
+import { firestore } from "firebase-admin";
+import * as functions from "firebase-functions";
+import { Config, asBoolean, isEnabled, asNumber } from "./config";
+import { computeCurrentCampaign } from "./compute-current-campaign";
+
+const db = firestore();
+const config = functions.config() as Config;
+const requireUniqueVote = asBoolean(
+  config.features.voting_campaigns.require_unique_vote
+);
+
+export const hasAlreadyVoted = functions.https.onCall(
+  async (_, context) => {
+    if (!requireUniqueVote) {
+      console.info("Config doesn't require unique vote per campaign; skipping...")
+      return false;
+    }
+    const voterEmail: string = context.auth!.token.email;
+    const currentDate = new Date();
+    const campaign = computeCurrentCampaign(currentDate, {
+      enabled: isEnabled(config.features.voting_campaigns),
+      startOn: asNumber(config.features.voting_campaigns.start_on),
+      endOn: asNumber(config.features.voting_campaigns.end_on)
+    });
```

Comment:
Not sure if I should do this or call the getCampaign function directly. The problem I see with calling getCampaign is that it might get slow